proxy/metrics_test.go

```diff
@@ -0,0 +1,65 @@
+package proxy_test
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/zalando/skipper/filters/builtin"
+	"github.com/zalando/skipper/metrics"
+	"github.com/zalando/skipper/metrics/metricstest"
+	"github.com/zalando/skipper/proxy/proxytest"
+	"github.com/zalando/skipper/routing"
+	"github.com/zalando/skipper/routing/testdataclient"
+)
+
+func TestMetricsUncompressed(t *testing.T) {
+	dm := metrics.Default
+	t.Cleanup(func() { metrics.Default = dm })
+
+	m := &metricstest.MockMetrics{}
+	metrics.Default = m
+
+	// will update routes after proxy address is known
+	dc := testdataclient.New(nil)
+	defer dc.Close()
+
+	p := proxytest.Config{
+		RoutingOptions: routing.Options{
+			FilterRegistry: builtin.MakeRegistry(),
+			DataClients:    []routing.DataClient{dc},
+		},
+	}.Create()
+	defer p.Close()
+
+	err := dc.UpdateDoc(fmt.Sprintf(`
+		test: Path("/test") -> setPath("/backend") -> "%s";
+		backend: Path("/backend") && Header("Accept-Encoding", "gzip") -> compress() -> inlineContent("backend response") -> <shunt>;
+	`, p.URL), nil)
+	require.NoError(t, err)
+
+	// wait for route update
+	time.Sleep(10 * time.Millisecond)
```

Comment:
I think you can also set `WaitFirstRouteLoad: true`

Reply:
That option is in skipper.Options, but you can likely use `./routing/routing.go:134:       SignalFirstLoad bool` and pass a channel or so. Maybe not in proxytest.